src/frequenz/sdk/actor/power_distributing/battery_pool.py

```diff
@@ -0,0 +1,122 @@
+                data.
+            max_blocking_duration_sec: This value tell what should be the maximum
+                timeout used for blocking failing component.
+        """
+        self._batteries = {
+            id: BatteryUtils(id, max_data_age_sec, max_blocking_duration_sec)
+            for id in battery_ids
+        }
+        self._run_method_called: bool = False
+
+    async def init(self) -> None:
+        """Run battery pool."""
+        tasks = [
+            asyncio.create_task(battery.init()) for battery in self._batteries.values()
+        ]
+        await asyncio.wait(tasks, return_when=ALL_COMPLETED)
+        self._run_method_called = True
+
+    def get_working_batteries(self, battery_ids: Set[int]) -> Set[int]:
+        """Get subset of battery_ids with working batteries.
+
+        Args:
+            battery_ids: batteries ids
+
+        Raises:
+            RuntimeError: If `run` method was not called at the beginning to
+                initialized object.
+            KeyError: If any battery in the given batteries is not in the pool.
+
+        Returns:
+            Subset of given batteries with working batteries.
+        """
+        if not self._run_method_called:
+            raise RuntimeError(
+                "`run` method not called or not awaited. Run it before first use"
+            )
+
+        ready: Set[int] = set()
+        blocked: Set[int] = set()
+        for bat_id in battery_ids:
+            if bat_id not in battery_ids:
+                ids = str(self._batteries.keys())
+                raise KeyError(f"No battery {bat_id} in pool. All batteries: {ids}")
+            if self._batteries[bat_id].is_ready():
+                ready.add(bat_id)
+            if self._batteries[bat_id].is_blocked():
+                blocked.add(bat_id)
+
+        working = ready - blocked
+
+        if len(working) > 0:
+            return working
+
+        _logger.warning("All batteries in %s are not working. Use blocked batteries.")
+        return ready & blocked
+
+    def update_last_request_status(self, result: Result):
+        """Update batteries in pool based on the last result from the request.
+
+        Args:
+            result: Summary of what batteries failed and succeed in last request.
+
+        Raises:
+            RuntimeError: If `run` method was not called at the beginning to initialize
+                object.
+        """
+        if not self._run_method_called:
+            raise RuntimeError(
+                "`run` method not called or not awaited. Run it before first use"
+            )
+
+        if isinstance(result, Success):
+            for bat_id in result.used_batteries:
+                self._batteries[bat_id].unblock()
+
+        elif isinstance(result, PartialFailure):
+            for bat_id in result.failed_batteries:
+                duration = self._batteries[bat_id].block()
+                if duration > 0:
+                    _logger.info(
```

Comment:
I'd suggest setting the log level to `warn` for this message.

Reply:
Done